### PR TITLE
Remove dependency on CMFDefault

### PR DESCRIPTION
**Pull request by tomgross**

### CODE REVIEW

**Comment by gforcada** [comment]
Testing it on http://jenkins.plone.org/job/pull-request/76

Not sure if we want to merge this on master *right now* as we are already on beta... /cc @tisto @esteele

**Comment by gforcada** [comment]
Forgot to say that great to see a package dependency being removed!

**Comment by gforcada** [comment]
You can check which packages are using CMF packages here also: http://jenkins.plone.org/view/Dependencies/job/plone-package-dependencies-cmf/

**Comment by davisagli** [comment]
I think we can merge this work in beta -- we still have time to fix any regressions -- but we should make sure that Products.CMFDefault is still present in the python environment in case addons still use it, so that they aren't forced to do more work to be Plone 5 compatible. Maybe there's a way we can make importing from it show a warning.

I'm a bit confused by the changes -- it looks like the PloneFolder module is removed, but is still referenced from Portal.py. Do you have a .pyc file lying around?

**Comment by gforcada** [comment]
@tomgross there are quite a few test failures on other packages: http://jenkins.plone.org/job/pull-request/76/console

Maybe is worth creating a cfg file on plips folder in buildout.coredev to keep all needed related branches together.

Creating a jenkins job for it is really easy, let us know about it (create a ticket on https://github.com/plone/jenkins.plone.org)

File: Products/CMFPlone/PloneTool.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
                         ManagePortal, ManageUsers, ModifyPortalContent, View
 from Products.CMFCore.interfaces import IDublinCore, IMutableDublinCore
 from Products.CMFCore.WorkflowCore import WorkflowException
-from Products.CMFDefault.DublinCore import DefaultDublinCoreImpl
+from .DublinCore import DefaultDublinCoreImpl
```
Comment (mauritsvanrees) [inline Products/CMFPlone/PloneTool.py]: Minor thing: plone.api convention says: do not use relative imports:
http://ploneapi.readthedocs.org/en/latest/contribute/conventions.html#about-imports

**Comment by tomgross** [comment]
There is a config already: https://github.com/plone/buildout.coredev/blob/5.0/plips/plip-nocmfdefault.cfg

**Comment by gforcada** [comment]
And there is a job as well: http://jenkins.plone.org/view/PLIPs/job/plip-noncmfdefault/

File: Products/CMFPlone/PloneTool.py (hunk 1 above)
Comment (gforcada) [inline Products/CMFPlone/PloneTool.py]: :+1:

**Comment by tisto** [comment]
@tomgross Awesome! Thanks for working on that! +1 for a merge if @esteele is ok with that.

File: Products/CMFPlone/PloneTool.py (hunk 1 above)
Comment (tisto) [inline Products/CMFPlone/PloneTool.py]: +1

**Comment by tisto** [comment]
Tests pass: http://jenkins.plone.org/job/plip-noncmfdefault/2/

**Comment by esteele** [comment]
Absolutely. Nice work!

**Comment by tisto** [comment]
The plip job somehow does not checkout the correct branch. I get quite a few test failures when I check out the branch manually.

**Comment by tisto** [comment]
@tomgross I fixed the plip job configuration. Your branch was not fetched at all. There are quite a few test failures in CMFPlone: 

http://jenkins.plone.org/view/PLIPs/job/plip-noncmfdefault/3/console

**Comment by tomgross** [comment]
readded accidentally deleted PloneFolder. All tests pass on my machine now.

**Comment by tisto** [comment]
Tests:

http://jenkins.plone.org/job/pull-request/79/
http://jenkins.plone.org/view/PLIPs/job/plip-noncmfdefault/4/

File: Products/CMFPlone/PloneTool.py (hunk 1 above)
Comment (tomgross) [inline Products/CMFPlone/PloneTool.py]: There are relative imports already https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/__init__.py

**Comment by tisto** [comment]
@tomgross both jobs show two failing tests.

**Comment by tomgross** [comment]
The branch was not in sync with master. I fixed it meanwhile.

**Comment by gforcada** [comment]
Let's give it a try then: http://jenkins.plone.org/job/pull-request/80/console

File: Products/CMFPlone/PloneTool.py (hunk 1 above)
Comment (mauritsvanrees) [inline Products/CMFPlone/PloneTool.py]: Not anymore. ;-) I fixed those just now on master at f4322dc20bffb7c9bb5ed271f2e9bc4b5c40e560.
And to put my money/time where my mouth is, I fixed the new ones on this branch as well, at 8c0ad6cfa63256d58a4a02feabcedf0df60b39f1.

**Comment by tisto** [comment]
$ grep -R --include=*.py 'Products.CMFDefault' src/

```
src//plone.app.content/plone/app/content/container.py:from Products.CMFDefault.DublinCore import DefaultDublinCoreImpl
src//plone.app.content/plone/app/content/item.py:from Products.CMFDefault.DublinCore import DefaultDublinCoreImpl
src//plone.app.content/setup.py:        'Products.CMFDefault',
src//plone.app.contentrules/setup.py:        'Products.CMFDefault',
src//plone.app.contenttypes/plone/app/contenttypes/migration/browser.py:from Products.CMFDefault.DublinCore import DefaultDublinCoreImpl
src//plone.app.contenttypes/plone/app/contenttypes/setuphandlers.py:from Products.CMFDefault.utils import bodyfinder
src//plone.app.controlpanel/plone/app/controlpanel/filter.py:from Products.CMFDefault.formlib.schema import SchemaAdapterBase
src//plone.app.controlpanel/plone/app/controlpanel/language.py:from Products.CMFDefault.formlib.schema import SchemaAdapterBase
src//plone.app.controlpanel/plone/app/controlpanel/ram.py:from Products.CMFDefault.formlib.schema import SchemaAdapterBase
src//plone.app.controlpanel/plone/app/controlpanel/skins.py:from Products.CMFDefault.formlib.schema import SchemaAdapterBase
src//plone.app.controlpanel/setup.py:        'Products.CMFDefault',
src//plone.app.form/setup.py:          'Products.CMFDefault',
src//plone.app.testing/plone/app/testing/layers.py:        ('Products.CMFDefault',                  {'loadZCML': True}, ),
src//plone.theme/plone/theme/interfaces.py:    from Products.CMFDefault.interfaces import ICMFDefaultSkin as IDefaultBrowserLayer
src//plone.theme/plone/theme/layer.py:    from Products.CMFDefault.interfaces import ICMFDefaultSkin
src//Products.ATContentTypes/Products/ATContentTypes/content/document.py:from Products.CMFDefault.utils import SimpleHTMLParser
src//Products.ATContentTypes/Products/ATContentTypes/setuphandlers.py:from Products.CMFDefault.utils import bodyfinder
src//Products.ATContentTypes/Products/ATContentTypes/tool/metadata.py:from Products.CMFDefault.MetadataTool import MetadataTool as BaseTool
src//Products.ATContentTypes/setup.py:          'Products.CMFDefault',
```

**Comment by gforcada** [comment]
@tisto we have this job that keeps track of all of them: http://jenkins.plone.org/view/Dependencies/job/plone-package-dependencies-cmf

**Pull request merged**